### PR TITLE
Exclude custom formatting commands

### PR DESCRIPTION
**Pull request by jkimbo**
terminal-quest has some custom formatting strings in it's code base which were being erroneously translated by the `fake_locale` script. This PR adds logic to ignore those formatting definitions.

N.B. I also restructured the script and added some basic tests to cover the change.

### CODE REVIEW

File: dev_locale/fake_locale [hunk 1]
```diff
@@ -12,225 +12,58 @@
...
-    print "Usage: fake-locale messages.pot en_QQ.po"
+from dev_locale.translate import trans_str  # noqa
 
-inFile = sys.argv[1]
-outFile = sys.argv[2]
 
-# load input template
-po = polib.pofile(inFile)
-new = polib.POFile()
+def main():
```
Comment (tombettany) [inline dev_locale/fake_locale]: Is this only ever run from the CLI?

File: dev_locale/fake_locale [hunk 2]
```diff
@@ -12,225 +12,58 @@
...
+    global tags
```
Comment (tombettany) [inline dev_locale/fake_locale]: There seems to be some confusion between the `tags` variable here and the one in the other file - it doesn't look like  `tags` is even used outside `main()` here.

File: dev_locale/translate.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
+# coding:utf8
```
Comment (tombettany) [inline dev_locale/translate.py]: Please add our standard copyright notice to the header:

```
#
# <filename>
#
# Copyright (C) 2016 Kano Computing Ltd.
# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
#
# <file_description>
# 
```

**Comment by tombettany** [comment]
Also, `__init__.py` files should contain the copyright notice too:

```python
#
# __init__.py
#
# Copyright (C) 2016 Kano Computing Ltd.
# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
#

__author__ = 'Kano Computing Ltd.'
__email__ = 'dev@kano.me'
```

**Comment by tombettany** [comment]
The main problem seems to be the `tags` issue, please either clarify or fix.

File: dev_locale/fake_locale [hunk 4]
```diff
@@ -12,225 +12,58 @@
...
-tags = False
-if '-tags' in sys.argv:
-    del sys.argv[sys.argv.index('-tags')]
-    tags = True
+if __name__ == '__main__' and __package__ is None:
+    os.sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
-if len(sys.argv) < 3:
-    print "Usage: fake-locale messages.pot en_QQ.po"
+from dev_locale.translate import trans_str  # noqa
```
Comment (tombettany) [inline dev_locale/fake_locale]: `noqa`?

**Comment by Ealdwulf** [comment]
Since this is a special case, it might be best to add an option to do it rather than adding to the default case. Otherwise looks good :-)

File: dev_locale/fake_locale (hunk 1 above)
Comment (jkimbo) [inline dev_locale/fake_locale]: Good point! Fixed.

File: dev_locale/fake_locale (hunk 2 above)
Comment (jkimbo) [inline dev_locale/fake_locale]: Yep good catch, missed this while refactoring. `tags` is now passed into `trans_str` as a param.

File: dev_locale/translate.py (hunk 3 above)
Comment (jkimbo) [inline dev_locale/translate.py]: 👍

File: dev_locale/fake_locale (hunk 4 above)
Comment (jkimbo) [inline dev_locale/fake_locale]: pep8 will complain with a `module level import not at top of file [E402]` unless the line is ignored

**Comment by jkimbo** [comment]
After chatting with @Ealdwulf decided that it's not worth adding a flag to enable this since it's unlikely to conflict with any actual strings. When you are happy @tombettany I'll merge.

**Comment by tombettany** [comment]
@jkimbo Fine with me. Feel free to merge

**Pull request merged**